src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -706,6 +717,9 @@ private void CheckSize(int countToAdd)
             }
         }
 
+#if NET46
+        [SecuritySafeCritical]
```

Comment:
I see your point.  The `LimitArrayPoolWriteStream` class is only defined in the .NET Core version of HttpContent and is used only in the .NET Core version of HttpClient.cs.

So, I think I will change this so that the `LimitArrayPoolWriteStream` implementation is not in the NET46 build (then I won't need the security annotation for that) and I'll update the HttpContent.cs file so it is has a NET46 version for those methods that is 100% same as Desktop.